frontend/src/metabase/containers/DataPicker/DataPickerContainer.tsx

```diff
@@ -114,7 +115,7 @@ function DataPicker({
   );
 
   useOnMount(() => {
-    if (dataTypes.length === 1) {
+    if (dataTypes.length === 1 && value.type !== dataTypes[0].id) {
```

Comment:
Fixes the case when `DataPicker` overwrites the initial value.

Reply:
Can we add a test for this?